src/ui/buttons/buttonDesigns/resizeButtonShowLabelText.jsx

```diff
@@ -0,0 +1,196 @@
+import type { ContentOptions } from './types';
+
+type AnimationProps = {|
+    designContainer : Object,
+    paypalLabelContainerElement : Object,
+    labelFontSize : number,
+    marginLabelContainer : number,
+    buttonHeight : number,
+    topPositionBlueLayer : string
+|};
+
+export function resizeButtonShowLabelTextComponent({ designLabelText, logoColor } : ContentOptions) : ChildType {
+    // experimentName must match elmo experiment name
+    const ANIMATION_LABEL = DESIGN_SMALL_BUTTON_CONFIG.cssClasses.ANIMATION_LABEL_CONTAINER;
+    const ANIMATION_CONTAINER = DESIGN_SMALL_BUTTON_CONFIG.cssClasses.ANIMATION_CONTAINER;
+    return (
+        <Fragment>
+            <PPLogo logoColor={ logoColor } />
+            <div class='blue-layer' />
+            <div class={ ANIMATION_LABEL } data-design-experiment='104519'>
+                <span>{designLabelText}</span>
+            </div>
+            <style innerHTML={ `
+                    .${ ANIMATION_CONTAINER } .blue-layer {
+                        position: fixed;
+                        opacity: 0;
+                    }
+                    .${ DESIGN_SMALL_BUTTON_CONFIG.cssClasses.DOM_READY } .${ ANIMATION_CONTAINER } img.${ DESIGN_SMALL_BUTTON_CONFIG.cssClasses.PAYPAL_LOGO }-pp{
+                        position: fixed;
+                        opacity:0;
+                    }
+                    .${ ANIMATION_CONTAINER } .${ ANIMATION_LABEL } {
+                        opacity: 0; 
+                        color: white;
+                        font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
+                    }
+                ` } />
+        </Fragment>
+    );
+}
+
+// Returns label container if the button sizes match
+export const resizeButtonShowLabelTextProps = function(document : Object, configuration : Object) : AnimationProps | null {
+    let labelFontSize = 8;
+    const { ANIMATION_CONTAINER, ANIMATION_LABEL_CONTAINER, PAYPAL_BUTTON_LABEL } = configuration.cssClasses;
+    const { min, smallMax, mediumMax } = configuration;
+    // get the animation main container to force specificity( in css ) and make sure we are running the right animation
+    const designContainer = (document && document.querySelector(`.${ ANIMATION_CONTAINER }`)) || null;
+    if (!designContainer) {
+        return null;
+    }
+
+    // return null if animation should not be played for the button size
+    const designContainerWidth = designContainer.offsetWidth;
+    if (designContainerWidth < min || designContainerWidth > mediumMax) {
+        // remove label element from dom
+        designContainer.querySelector(`.${ ANIMATION_LABEL_CONTAINER }`).remove();
+        return null;
+    }
+
+    const topPositionBlueLayer = designContainerWidth === mediumMax ? '26%' : '25%';
+
+    if (designContainerWidth >= smallMax) {
+        labelFontSize =  designContainerWidth === mediumMax ? 12 : 11;
+    }
+    
+    let buttonHeight = designContainer.offsetHeight;
```

Comment:
What is this doing? And is there a more elegant way than this?

Reply:
I am afraid there is not, in here we are just getting the button's height